### PR TITLE
Implement machine and VM deletion in terraform

### PR DESCRIPTION
**Pull request by krousey**
**What this PR does / why we need it**:
This enables the terraform controller to delete nodes.

### CODE REVIEW

File: cloud/terraform/cmd/terraform-machine-controller/Dockerfile [hunk 1]
```diff
@@ -24,6 +24,18 @@ RUN CGO_ENABLED=0 GOOS=linux go install -a -ldflags '-extldflags "-static"' sigs
 
 # Final container
 FROM alpine:3.7
-RUN apk --no-cache add --update ca-certificates bash openssh terraform
-RUN echo 'plugin_cache_dir = "$HOME/.terraform.d/plugin-cache"' >> ~/.terraformrc
+
+ENV TERRAFORM_VERSION=0.11.7
```
Comment (krousey) [inline cloud/terraform/cmd/terraform-machine-controller/Dockerfile]: The apk version was 0.11.0, and `terraform destroy` didn't have an auto-approve flag.

**Comment by krousey** [comment]
/hold
Will build a push controller image before merge.

File: cloud/terraform/machineactuator.go [hunk 2]
```diff
@@ -345,7 +345,42 @@ func runTerraformCmd(stdout bool, workingDir string, arg ...string) (bytes.Buffe
 
 func (tf *TerraformClient) Delete(machine *clusterv1.Machine) error {
 	glog.Infof("TERRAFORM DELETE.\n")
```
Comment (karan) [inline cloud/terraform/machineactuator.go]: Remove this log line?

File: cloud/terraform/machineactuator.go [hunk 3]
```diff
@@ -345,7 +345,42 @@ func runTerraformCmd(stdout bool, workingDir string, arg ...string) (bytes.Buffe
...
+
+	if instance == nil {
+		return nil
+	}
+
+	homedir, err := getHomeDir()
+	if err != nil {
+		return err
+	}
+	machinePath := path.Join(homedir, fmt.Sprintf(".terraform.d/kluster/machines/%s", machine.ObjectMeta.Name))
```
Comment (karan) [inline cloud/terraform/machineactuator.go]: nite but we're joining this path in create as well. Can you just create a helper like `getMachinePath`?

File: cloud/terraform/machineactuator.go [hunk 4]
```diff
@@ -406,7 +441,7 @@ func (tf *TerraformClient) updateMasterInPlace(goalMachine *clusterv1.Machine) e
 	// Control plane upgrade
 	if goalMachineControlPlaneVersion != currentControlPlaneVersion {
 		// Pull the kudeadm for target version K8s.
-		cmd := fmt.Sprintf("curl -sSL https://dl.k8s.io/release/v%s/bin/linux/amd64/kubeadm | sudo tee /usr/bin/kubeadm > /dev/null; " +
+		cmd := fmt.Sprintf("curl -sSL https://dl.k8s.io/release/v%s/bin/linux/amd64/kubeadm | sudo tee /usr/bin/kubeadm > /dev/null; "+
```
Comment (karan) [inline cloud/terraform/machineactuator.go]: Why the change here?

File: cloud/terraform/machineactuator.go (hunk 3 above)
Comment (krousey) [reply cloud/terraform/machineactuator.go]: I would but I expect this to be deleted eventually once we stop using disk for storing state.

File: cloud/terraform/machineactuator.go (hunk 4 above)
Comment (krousey) [reply cloud/terraform/machineactuator.go]: gofmt. My editor runs it on every save and it does this.

File: cloud/terraform/machineactuator.go (hunk 2 above)
Comment (krousey) [reply cloud/terraform/machineactuator.go]: Done.

**Comment by krousey** [comment]
Address comments and also bumped the tag everywhere

**Comment by krousey** [comment]
/hold cancel
Image built and pushed
